core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -378,11 +375,13 @@ private[log] class Cleaner(val id: Int,
    * @param segments The group of segments being cleaned
    * @param map The offset map to use for cleaning segments
    * @param deleteHorizonMs The time to retain delete tombstones
+   * @param stats Collector for cleaning statistics
    */
   private[log] def cleanSegments(log: Log,
                                  segments: Seq[LogSegment], 
                                  map: OffsetMap, 
-                                 deleteHorizonMs: Long) {
+                                 deleteHorizonMs: Long,
+                                 stats: CleanerStats = new CleanerStats()) {
```

Comment:
Ack. I thought it seemed odd as an exposed method, so this was a weak attempt to conceal the oddness. I guess it's only exposed for testing though, so maybe it's not too bad.